src/type_seed.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    type_seed(x)
+
+Computes a value to use as a seed for computing the hash value of a type.
```

Comment:
@NHDaly could you have someone take a look at the implementation in this file? It makes sense to me, but I'm not sure if there are edges/assumptions/things I'm not aware of here

Reply:
@comnik Could you please look at this and give your thoughts?

Reply:
/cc @comnik

Reply:
@sacha0 is the most expert expert on this topic that I know of, though he isn't working on this type of thing right now.

Reply:
I'll take a look tomorrow, sorry I wasn't able to follow the discussion here so far.

Reply:
The main thing we're looking for is your thoughts on this approach to computing a stable type seed.

Reply:
So I have to say I'm not really an expert in this, but I read the PR and the approach seems sensible to me. Being sensitive to the fully qualified name by default seems fine as well, as long as we have the ability to overwrite the type seed after refactorings.

For use in the RAI code base I am worried about the performance implications, see my two other comments.